core/src/main/java/com/googlecode/psiprobe/tools/url/UrlParser.java

```diff
@@ -34,6 +34,13 @@
 
   /**
    * Instantiates a new url parser.
+   */
+  public UrlParser() {
```

Comment:
Learning...ok sonar normally complains about this and I've been adding these for years.  Ran some quit tests, did a little more research, and now certainly see what this is written this way.  The default one is junk anyways even with my previous knowledge and would likely fail.  What I didn't know was that java wouldn't otherwise allow creation.  I assumed wrongly that it did and adding it was just for clarity sake.  So I'm going to run some more scans to sonar without to see where it complains and see if it doesn't provide a more noval solution.